src/all_implementations.jl

```diff
@@ -1585,6 +1586,7 @@ function is_valid(layout::IndexedOptionArray)
     return is_valid(layout.content)
 end
 
+Base.eltype(layout::IndexedOptionArray) = eltype(layout.content)
```

Comment:
Unlike `IndexedArray`, `IndexedOptionArray` elements can be `missing`.

```suggestion
Base.eltype(layout::IndexedOptionArray) = Union{Missing, eltype(layout.content)}
```